drivers/pds/lib/utils.go

```diff
@@ -1441,58 +1511,69 @@ func CreateTpccWorkloads(dataServiceName string, deploymentID string, scalefacto
 	return false, errors.New("TPCC run failed.")
 }
 
-// CreateDataServiceWorkloads func
-func CreateDataServiceWorkloads(dataServiceName string, deploymentID string, scalefactor string, iterations string, deploymentName string, namespace string) (*corev1.Pod, *v1.Deployment, error) {
+//CreateDataServiceWorkloads generates workloads for the given dataservices
+func CreateDataServiceWorkloads(params WorkloadGenerationParams) (*corev1.Pod, *v1.Deployment, error) {
 	var dep *v1.Deployment
 	var pod *corev1.Pod
 
-	dnsEndpoint, err := GetDeploymentConnectionInfo(deploymentID)
+	dnsEndpoint, err := GetDeploymentConnectionInfo(params.DeploymentID)
 	if err != nil {
-		log.Errorf("An Error Occured while getting connection info %v", err)
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("error occured while getting connection info %v", err)
 	}
 	log.Infof("Dataservice DNS endpoint %s", dnsEndpoint)
 
-	pdsPassword, err := GetDeploymentCredentials(deploymentID)
+	pdsPassword, err := GetDeploymentCredentials(params.DeploymentID)
 	if err != nil {
-		log.Errorf("An Error Occured while getting credentials info %v", err)
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("error occured while getting credentials info %v", err)
 	}
 
-	switch dataServiceName {
+	switch params.DataServiceName {
 	case postgresql:
-		dep, err = CreatepostgresqlWorkload(dnsEndpoint, pdsPassword, scalefactor, iterations, deploymentName, namespace)
+		dep, err = CreatepostgresqlWorkload(dnsEndpoint, pdsPassword, params.ScaleFactor, params.Iterations, params.DeploymentName, params.Namespace)
 		if err != nil {
-			log.Errorf("An Error Occured while creating postgresql workload %v", err)
-			return nil, nil, err
+			return nil, nil, fmt.Errorf("error occured while creating postgresql workload %v", err)
```

Comment:
can you log like this 
fmt.Errorf("error occured while creating postgresql workload, Err: %v", err)

it will be more clear

Reply:
done